stonesoup/architecture/__init__.py

```diff
@@ -0,0 +1,603 @@
+        one node with 0 recipients, and there exists a path to this node from every other node in
+        the architecture.
+        """
+        top_nodes = self.top_level_nodes
+        if len(top_nodes) != 1:
+            return False
+        else:
+            top_node = top_nodes.pop()
+        for node in self.all_nodes - self.top_level_nodes:
+            try:
+                _ = self.shortest_path_dict[node][top_node]
+            except KeyError:
+                return False
+        return True
+
+    @property
+    def is_connected(self):
+        return nx.is_connected(self.to_undirected)
+
+    @property
+    def to_undirected(self):
+        return self.di_graph.to_undirected()
+
+    def __len__(self):
+        return len(self.di_graph)
+
+    @property
+    def fully_propagated(self):
+        """Checks if all data for each node have been transferred
+        to its recipients. With zero latency, this should be the case after running propagate"""
+        for edge in self.edges.edges:
+            if len(edge.unsent_data) != 0:
+                return False
+
+        return True
+
+
+class NonPropagatingArchitecture(Architecture):
+    """
+    A simple Architecture class that does not simulate propagation of any data. Can be used for
+    performing network operations on an :class:`~.Edges` object.
+    """
+    def propagate(self, time_increment: float):
+        pass
+
+
+class InformationArchitecture(Architecture):
+    """The architecture for how information is shared through the network. Node A is "
+    "connected to Node B if and only if the information A creates by processing and/or "
+    "sensing is received and opened by B without modification by another node. """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if isinstance(self, InformationArchitecture):
+            for node in self.all_nodes:
+                if isinstance(node, RepeaterNode):
+                    raise TypeError("Information architecture should not contain any repeater "
+                                    "nodes")
+        for fusion_node in self.fusion_nodes:
+            pass  # fusion_node.tracker.set_time(self.current_time)
+
+    def measure(self, ground_truths: List[GroundTruthPath], noise: Union[bool, np.ndarray] = True,
+                **kwargs) -> Dict[SensorNode, Set[Union[TrueDetection, Clutter]]]:
+        """ Similar to the method for :class:`~.SensorSuite`. Updates each node. """
+        all_detections = dict()
+
+        # Get rid of ground truths that have not yet happened
+        # (ie GroundTruthState's with timestamp after self.current_time)
+        new_ground_truths = set()
+        for ground_truth_path in ground_truths:
+            # need an if len(states) == 0 continue condition here?
+            new_ground_truths.add(ground_truth_path.available_at_time(self.current_time))
```

Comment:
Possibly... I think this would take too much fiddling to be necessarily worth it, as we (I) then have to keep track of the last timestep we measured at in addition to the current time, and amend the ground_truth_path.available_at_time method. If you feel strongly that the current implementation should change I'd be happy to do so of course.